automod/testing.go

```diff
@@ -0,0 +1,142 @@
+package automod
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"log/slog"
+	"os"
+	"time"
+
+	appbsky "github.com/bluesky-social/indigo/api/bsky"
+	"github.com/bluesky-social/indigo/atproto/identity"
+	"github.com/bluesky-social/indigo/atproto/syntax"
+)
+
+func simpleRule(evt *RecordEvent, post *appbsky.FeedPost) error {
+	for _, tag := range post.Tags {
+		if evt.InSet("bad-hashtags", tag) {
+			evt.AddRecordLabel("bad-hashtag")
+			break
+		}
+	}
+	for _, facet := range post.Facets {
+		for _, feat := range facet.Features {
+			if feat.RichtextFacet_Tag != nil {
+				tag := feat.RichtextFacet_Tag.Tag
+				if evt.InSet("bad-hashtags", tag) {
+					evt.AddRecordLabel("bad-hashtag")
+					break
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func EngineTestFixture() Engine {
+	rules := RuleSet{
+		PostRules: []PostRuleFunc{
+			simpleRule,
+		},
+	}
+	cache := NewMemCacheStore(10, time.Hour)
+	flags := NewMemFlagStore()
+	sets := NewMemSetStore()
+	sets.Sets["bad-hashtags"] = make(map[string]bool)
+	sets.Sets["bad-hashtags"]["slur"] = true
+	dir := identity.NewMockDirectory()
+	id1 := identity.Identity{
+		DID:    syntax.DID("did:plc:abc111"),
+		Handle: syntax.Handle("handle.example.com"),
+	}
+	dir.Insert(id1)
+	engine := Engine{
+		Logger:    slog.Default(),
+		Directory: &dir,
+		Counters:  NewMemCountStore(),
+		Sets:      sets,
+		Flags:     flags,
+		Cache:     cache,
+		Rules:     rules,
+	}
+	return engine
+}
+
+func MustLoadCapture(capPath string) AccountCapture {
+	f, err := os.Open(capPath)
+	if err != nil {
+		panic(err)
+	}
+	defer func() { _ = f.Close() }()
+
+	raw, err := io.ReadAll(f)
+	if err != nil {
+		panic(err)
+	}
+
+	var capture AccountCapture
+	if err := json.Unmarshal(raw, &capture); err != nil {
+		panic(err)
+	}
+	return capture
+}
+
+// Test helper which processes all the records from a capture. Intentionally exported, for use in other packages.
+//
+// This method replaces any pre-existing directory on the engine with a mock directory.
+func ProcessCaptureRules(e *Engine, capture AccountCapture) error {
```

Comment:
Definitely feeling the lack of packages here or other strong organizational cues for what's testing-land and what's not.  Understood that that's a future PR topic, but just want to ratify that outloud :)

The comment being explicit that this function is rewiring the engine is definitely very very good and appreciated :+1: , because I wouldn't necessarily presume that from the signature or name otherwise.